transformer_lens/hook_points.py

```diff
@@ -523,6 +592,9 @@ def save_hook(tensor, hook, is_backward=False):
                 resid_stream = resid_stream[0]
 
             # for attention heads the pos dimension is the third from last
+            if hook.name is None:
+                raise RuntimeError("Hook should have been provided a name")
+
```

Comment:
There is a duplicate here with an assert and an if check. I prefer the if check, so if you can move that up to where the assert is. that would be great.